### DISCUSSION TITLE
GarnetServer.exe as Windows Service with garnet.conf

### DISCUSSION THREAD

**Discussion by mae2-s** [microsoft/garnet#481] [Q&A] [answered]
Hello everyone,
I want to run GarnetServer.exe as a WindowsService. Now it would be good if the service would use the garnet.conf file. I have already tried everything with cs (create service) under Windows. Unfortunately it did not work with the parameter ‘--config-import-path’. Since I am not a developer I do not want to create the service myself (although it would then be possible to customise it with the path). Therefore the question if it would be possible if the garnet.config is present in the same path as the .exe, it then also applies it.

**Comment by mae2-s** [comment, marked as answer]
with the tool NSSM it is possible to generate a service under Windows which uses parameters

**Reply by wessisaac** [reply]
Could you please share us the command to create Garnet service with configuration file garnet.conf

**Comment by mae2-s** [comment]
Hello, 
you need this Software https://nssm.cc/commands 
and command is
nssm.exe install GarnetService "D:\temp\garnet\GarnetServer.exe" --config-import-path "d:\temp\garnet\garnet.conf"

**Reply by wessisaac** [reply]
Yes, it is working. Thank you very much for your timely help. We couldn't configure the service with the Garnet.conf file using sc.exe, but it is working as expected with nssm.

Garnet as a service In Windows OS,

download 64 bit package from Git-Hub release (https://github.com/microsoft/garnet/releases)
Place the downloaded package in your convenient place. Eg: C:\Program Files\Garnet
In windows, Garnet as a service is best. To install garnet as a service, download 64 bit nssm from the url "https://nssm.cc/download" and place the nssm.exe folder into your Garnet package folder
In the command prompt enter the command, >nssm install Garnet
A GUI will open, in the path text box, you can browse your GarnetServer.exe path
Path: C:\Program Files\Garnet\GarnetServer.exe
In the argument text box, please enter the below command with path of your Garnet configuration file
Arguments: GarnetServer.exe --config-import-path "C:\Program Files\Garnet\Garnet.conf"
Then click install button
Now you can access Garnet in windows
Use Redis-cli in windows:
To use redis-cli in windows, download redis-windows package from the path "https://github.com/zkteco-home/redis-windows/releases/tag/7.4.2", then copy and paste the redis-cli.exe file (redis-clie.exe single file is enough) in Garnet package folder.
In command prompt, you can place the Garnet path, then redis-cli commands can be used.
